Fail to build libmng [AutotoolsPackage]

@alalazo I got this error while installing `libmng`:

```
...
==> Already patched libmng
==> Building libmng [AutotoolsPackage]
==> Executing phase : 'autoreconf'
==> Executing phase : 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/my/path/spack/var/spack/stage/libmng-2.0.2-2x5fkukzf3sf4uexegr3n35jwmy5pclu/libmng-2.0.2/configure' '--prefix=/my/path/spack/opt/spack/linux-scientificcernslc6-x86_64/gcc-6.2.0/libmng-2.0.2-2x5fkukzf3sf4uexegr3n35jwmy5pclu'
/my/path/spack/lib/spack/spack/build_systems/autotools.py:265, in configure:
     258      def configure(self, spec, prefix):
     259          """Runs configure with the arguments specified in :py:meth:`.configure_args`
     260          and an appropriately set prefix.
     261          """
     262          options = ['--prefix={0}'.format(prefix)] + self.configure_args()
     263  
     264          with working_dir(self.build_directory, create=True)
```

And this is the spack-build.out:

```
...
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
configure: error: source directory already configured; run "make distclean" there first
```

Before merge #2859 I could install it correctly but now I get this, do you think that could be related with the commit or maybe I have to install it in a different way now?

--- a/var/spack/repos/builtin/packages/libmng/package.py
+++ b/var/spack/repos/builtin/packages/libmng/package.py
@@ -42,3 +42,7 @@ class Libmng(AutotoolsPackage):
         # jpeg requires stdio to beincluded before its headrs.
         filter_file(r'^(\#include \<jpeglib\.h\>)',
                     '#include<stdio.h>\n\\1', 'libmng_types.h')
+
+    @run_before('configure')
+    def clean_configure_directory(self):
+        make('distclean')

